### PR TITLE
Feature/62 set exit status according to tacing result

### PR DESCRIPTION
**Pull request by redcatbear**
This is quite a large change for the feature. But I had to fix "the one" test case in order to get green again.

### CODE REVIEW

**Comment by redcatbear** [comment]
Still go one part of the test in TestCliStarter.java that I don't like:

Since the standard working directory for the tests is the project base directory, everything under "doc/" is used when no arguments are given. This leads to indeterministic test results since the documentation is part of tracing.

Would be good if we could reliably change the current working directory before the test.

File: src/test/java/openfasttrack/cli/TestCliStarter.java [hunk 1]
```diff
@@ -52,144 +57,243 @@
...
     {
-        return new String(Files.readAllBytes(file), StandardCharsets.UTF_8);
+        final Path file = this.outputFile;
+        try
+        {
+            return new String(Files.readAllBytes(file), StandardCharsets.UTF_8);
+        }
+        catch (final IOException e)
+        {
+            e.printStackTrace();
```
Comment (kaklakariada) [inline src/test/java/openfasttrack/cli/TestCliStarter.java]: Rethrow exception

File: src/main/java/openfasttrack/cli/commands/Performable.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * Interface for OpenFastTrack commands.
+ */
+public interface Performable
+{
+    /**
+     * Run an OpenFastTrack command
+     * 
+     * @return <code>true</code> if the operation performed was successful.
+     */
+    public boolean run();
```
Comment (kaklakariada) [inline src/main/java/openfasttrack/cli/commands/Performable.java]: `public` is default, could be omitted

Comment (redcatbear) [inline src/main/java/openfasttrack/cli/commands/Performable.java]: As we discussed I find it more expressive and consequent to add "public".

**Comment by redcatbear** [comment]
Did not use isEmptyString() because the test is against a ByteStream. Using toString() would yield "null". Testing for Zero stream size seems cleaner to me.

**Comment by kaklakariada** [comment]
@redcatbear please resolve the merge conflicts ;)

**Pull request merged**